components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -468,14 +471,18 @@ private OAuth2IntrospectionResponseDTO validateAccessToken(OAuth2TokenValidation
         return introResp;
     }
 
-    private String getAuthzUser(AccessTokenDO accessTokenDO) throws IdentityOAuth2Exception {
+    private String getAuthzUser(AccessTokenDO accessTokenDO, boolean isAuthFlow) throws IdentityOAuth2Exception {
 
         AuthenticatedUser user = accessTokenDO.getAuthzUser();
 
         if (user.isFederatedUser()) {
             return user.getAuthenticatedSubjectIdentifier();
         }
 
+        if (isAuthFlow) {
```

Comment:
Why do we need to this?
I guess that deserves a comment in the code

Reply:
Added comments